api/lorawan.proto

```diff
@@ -272,6 +279,8 @@ message TxSettings {
   // Time of the gateway when the uplink message was received, or when the downlink message should be transmitted.
   // For downlink, this requires the gateway to have GPS time synchronization.
   google.protobuf.Timestamp time = 11 [(gogoproto.stdtime) = true];
+  // Additional information for the downlink from the original TxRequest; necessary for the BasicStation LNS gateways.
+  RequestInfo request_info = 12;
```

Comment:
Why is this in a separate message? Can't we just add the fields to `TxSettings` directly, like we do with the other downlink-only `tx_power` field?

Reply:
Is there any advantage to adding them directly? Or is that a convention we follow?
I've used separate message only to group all the items necessary only for `BasicStation`.

Reply:
Firstly, we don't tailor our API for specific gateways. Secondly, `antenna_index` may also be relevant for other multi-antenna gateways. 

The `rx_window` and `class` fields only seem relevant for gateways that have their own scheduler, so I don't quite understand why they need to be in the API. 

From the [comment leading to this PR](https://github.com/TheThingsNetwork/lorawan-stack/issues/120#issuecomment-476235240) I understand that we need to come up with an artificial `xtime`. Why do we need to know the RX window for that? Isn't it sufficient to have the Basic Station frontend set the chosen frequency/DR as RX1 (regardless), an `RxDelay` of 15 and `xtime` of `Timestamp - 15sec` (accounting for overflow)?